src/lib.rs

```diff
@@ -90,19 +109,20 @@ async fn create_event(event: web::Json<Event>, state: web::Data<State>) -> HttpR
     HttpResponse::Created().finish()
 }
 
-#[get("/calendar/{id}")]
-async fn get_calendar(id: web::Path<String>, state: web::Data<State>) -> HttpResponse {
-    info!("getting calendar for {}", id);
+#[get("/calendar/{username}")]
+async fn get_calendar(username: web::Path<String>, state: web::Data<State>) -> HttpResponse {
+    info!("getting calendar for {}", username);
 
     let mut stmt_result = (&state).conn.lock().expect("failed to lock conn");
     let conn = &mut *stmt_result;
     let mut stmt = conn
-        .prepare("SELECT name, calendar_id, date_time FROM event")
+        .prepare("SELECT name, calendar_id, date_time FROM event WHERE username = ?1")
         .unwrap();
 
     let event_iter = stmt
-        .query_map([], |row| {
+        .query_map(&[username.as_str()], |row| {
             Ok(Event {
+                username: "".parse().unwrap(),
```

Comment:
Is this an empty string? Maybe it's better to actually set the username to an actual value?